runtime/core/exec_aten/testing_util/tensor_factory.h

```diff
@@ -1019,7 +1032,14 @@ class TensorFactory {
               data_.data(),
               dim_order_.data(),
               strides_.data(),
-              dynamism) {}
+              dynamism) {
+      // The only valid values for bool are 0 and 1; coerce!
+      if constexpr (std::is_same_v<true_ctype, bool>) {
```

Comment:
It seems in `ifdef aten` section this transformation is unconditional while here it is applied only if it is bool. I dont fully understand.

Reply:
it doesn't build with bits16 and we don't have bit_cast, otherwise I would make it unconditional here and use bit_cast.

Reply:
I am guessing bit_cast not being usable is related to it available only in c++20.

Reply:
yes; we will have access to c10::bit_cast once I get back to code sharing but currently we don't have it.